src/app/digital/utils/ExpressionParser/GenerateTokens.ts

```diff
@@ -0,0 +1,93 @@
+import {SubEquals} from "core/utils/StringUtils";
```

Comment:
In general see if any of these `for` loops can be reduced to like `find`, `findIndex`, `map`, etc.